src/platform/iotjs_module_i2c-linux-general.inl.h

```diff
@@ -0,0 +1,243 @@
+/* Copyright 2016 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#ifndef IOTJS_MODULE_I2C_LINUX_GENERAL_INL_H
+#define IOTJS_MODULE_I2C_LINUX_GENERAL_INL_H
+
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <linux/i2c-dev.h>
+
+
+#include "module/iotjs_module_i2c.h"
+
+
+namespace iotjs {
+
+
+int fd;
+int8_t addr;
+
+
+class I2cLinuxGeneral : public I2c {
+ public:
+  explicit I2cLinuxGeneral(JObject& ji2c);
+
+  static I2cLinuxGeneral* GetInstance();
+
+  virtual int SetAddress(int8_t address);
+  virtual int Scan(I2cReqWrap* i2c_req);
+  virtual int Open(I2cReqWrap* i2c_req);
+  virtual int Close();
+  virtual int Write(I2cReqWrap* i2c_rea);
+  virtual int WriteByte(I2cReqWrap* i2c_rea);
+  virtual int WriteBlock(I2cReqWrap* i2c_rea);
+  virtual int Read(I2cReqWrap* i2c_rea);
+  virtual int ReadByte(I2cReqWrap* i2c_rea);
+  virtual int ReadBlock(I2cReqWrap* i2c_rea);
+};
+
+
+I2cLinuxGeneral::I2cLinuxGeneral(JObject& ji2c)
+    : I2c(ji2c) {
+}
+
+
+I2cLinuxGeneral* I2cLinuxGeneral::GetInstance() {
+  return static_cast<I2cLinuxGeneral*>(I2c::GetInstance());
+}
+
+
+JObject makeI2cArray(char* data, int len) {
+  // FIXME: Need to use JS array. (Not implemented)
```

Comment:
JS arrays are implemented in JerryScript and there API functions to handle them. What is the problem here?